src/redux/rockets/rockets-slice.js

```diff
@@ -1 +1,24 @@
-// rockets component
+// rocket-slice.js
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const fetchRockets = createAsyncThunk('rocket/fetchRockets', async () => {
+  const response = await axios.get('https://api.spacexdata.com/v3/rockets');
+  return response.data.map(({
+    id, rocket_name: name, rocket_type: type, flickr_images: flickrImages,
+  }) => ({
+    id, name, type, flickrImages,
+  }));
+});
+
+export const rocketSlice = createSlice({
+  name: 'rocket',
+  initialState: { rockets: [] },
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchRockets.fulfilled, (state, action) => ({ ...state, rockets: action.payload }));
+  },
+});
```

Comment:
With the implementation of error handling in `fetchRockets` 💪🏼, this presents a valuable opportunity to enhance the `extraReducers` in `rocketSlice` by adding additional cases for the pending and rejected scenarios.

To accommodate these changes, consider introducing two new properties, **isLoading** and **error**, to your `rocketSlice`. These properties will serve to effectively manage the loading state and store any potential errors encountered during data fetching 🎯 .